### PR TITLE
Replace tarpaulin in CI and update toolchain to latest nightly

### PR DESCRIPTION
**Pull request by simonsso**
- replace tarpaulin with cargo llvm-lcov
- update toolchain to latest nightly
- AI clean up `cargo clippy --fix`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -87,16 +86,15 @@ jobs:
...
-      - name: Run cargo-tarpaulin
-        uses: actions-rs/tarpaulin@v0.1
-        with:
-          version: ${{ env.tarpaulin-vers }}
-          args: "--avoid-cfg-tarpaulin --all-features --workspace --timeout 120 --exclude runtimes-eden nodle-parachain --exclude-files **/mock.rs **/weights.rs **/migrations.rs"
-
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Test and gather coverage
+        run: cargo llvm-cov --lcov --output-path lcov.info --all-features --workspace --timeout 120 --exclude runtimes-eden nodle-parachain --exclude-files **/mock.rs **/weights.rs **/migrations.rs"
```
Comment (simonsso) [inline .github/workflows/ci.yml:92]:
```suggestion
        run: cargo llvm-cov --lcov --output-path lcov.info --all-features --workspace --timeout 120 --exclude runtimes-eden nodle-parachain --exclude-files **/mock.rs **/weights.rs **/migrations.rs
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -87,16 +86,15 @@ jobs:
...
-
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Test and gather coverage
+        run: cargo llvm-cov --lcov --output-path lcov.info --all-features --workspace --exclude runtimes-eden --exclude nodle-parachain  
       - name: Upload to Codecov
         uses: codecov/codecov-action@v3.0.0
         with:
           fail_ci_if_error: false
+          files: lcov.info
```
Comment (aliXsed) [inline .github/workflows/ci.yml:97]: @simonsso looks like there is an issue here that the codecov fails to show the PR correctly.

Comment (simonsso) [reply .github/workflows/ci.yml:97]: I will check, but it looked like clippy was unhappy when I changed the compiler to a recent nightly

File: node/src/command.rs [hunk 3]
```diff
@@ -225,14 +225,11 @@ pub fn run() -> Result<()> {
...
-						"Compile with --features=runtime-benchmarks \
+				BenchmarkCmd::Storage(_) => Err(sc_cli::Error::Input(
+					"Compile with --features=runtime-benchmarks \
 						to enable storage benchmarks."
-							.into(),
-					)
-					.into())
-				}
+						.into(),
+				)),
```
Comment (simonsso) [inline node/src/command.rs:232]: This change looks reasonable, the other arms are not using explicit return and at least one returns `Err()` already

File: pallets/grants/src/tests.rs
Comment (simonsso) [inline pallets/grants/src/tests.rs:1]: Automatic from clippy --fix

File: pallets/allocations/src/tests.rs [hunk 4]
```diff
@@ -699,3 +699,8 @@ fn no_issuance() {
 		);
 	})
 }
+
+#[test]
+fn test_releases_default() {
+	assert_eq!(Releases::default(), Releases::V0);
+}
```
Comment (simonsso) [inline pallets/allocations/src/tests.rs:706]: Used this to verify clippy's work

**Comment by aliXsed** [comment]
@simonsso approved. I'd say the codecov failure is now expected as you have changed our codecov tool and the new tool may report slightly less coverage.

**Review by aliXsed — APPROVED** [review]

**Pull request merged**